docs/components/query.md

```diff
@@ -1,65 +1,93 @@
 * Metrics received from Prometheus remote write streams (see [Receiver](receive.md))
-* Another Querier (you can stack Queriers on top of each other)
-* Non-Prometheus systems!
-  * e.g [OpenTSDB](../integrations.md#opentsdb-as-storeapi)
+* Another Querier (you can stack Queriers on top of each other!)
+* Non-Prometheus systems:
+  * e.g [OpenTSDB](../integrations.md#opentsdb-as-storeapi) or [RemoteRead](https://github.com/G-Research/thanos-remote-read)
 
-Thanks to that, you can run queries (manually, from Grafana or via Alerting rule) that aggregate metrics from mix of those sources.
+Thanks to that, you can run queries through API, Thanos UI, Grafana UI or via alerting or recording rules that aggregate metrics from mix of those sources.
 
 Some examples:
 
-* `sum(cpu_used{cluster=~"cluster-(eu1|eu2|eu3|us1|us2|us3)", job="service1"})` that will give you sum of CPU used inside all listed clusters for service `service1`. This will work even if those clusters runs multiple Prometheus servers each. Querier will know which data sources to query.
+* `sum(cpu_used{cluster=~"cluster-(eu1|eu2|eu3|us1|us2|us3)", job="service1"})` might give you sum of CPU used inside all listed clusters for service `service1`. This will work even if those clusters run multiple Prometheus servers each. Querier will know which data sources to query.
 
-* In single cluster you shard Prometheus functionally or have different Prometheus instances for different tenants. You can spin up Querier to have access to both within single Query evaluation.
+* In single cluster you shard Prometheus functionally or have different Prometheus instances for different tenants. You can spin up Querier to have access to all of the instances within single Query evaluation.
 
-### Run-time deduplication of HA groups
+### Use Case 2: Run-time deduplication of HA groups
 
-Prometheus is stateful and does not allow replicating its database. This means that increasing high availability by running multiple Prometheus replicas is not very easy to use. Simple load balancing will not work as for example after some crash, replica might be up but querying such replica will result in small gap during the period it was down. You have a second replica that maybe was up, but it could be down in other moment (e.g rolling restart), so load balancing on top of those is not working well.
+Prometheus is stateful and by design does not allow replicating its database. This means that increasing high availability by running multiple Prometheus replicas is not very easy to use. Simple load balancing will not work as for example after some crash, a replica might be up but querying such replica will result in small gap during the period it was down. You have a second replica that maybe was up, but it could be down in other moment (e.g. rolling restart), so requests based load-balancing on top of those two or more instances won't be accurate.
```

Comment:
> This means that increasing high availability by running multiple Prometheus replicas is not very easy to use

Not very easy to use in what respect? Perhaps we can reword this to something like  

```
 well.
Prometheus is stateful and by design does not allow replicating its database. Therefore increasing high availability by running multiple Prometheus replicas can introduce some challenges from a querying perspective.
```

Reply:
sgtm!